src/blueprint.zig

```diff
@@ -105,9 +108,47 @@ pub const Blueprint = struct {
 		}
 		return .{.success = self};
 	}
+
+	pub const PasteMode = enum {all, degradable};
+
+	pub fn pasteInGeneration(self: Blueprint, pos: Vec3i, chunk: *ServerChunk, mode: PasteMode) void {
+		switch(mode) {
+			.all => _pasteInGeneration(self, pos, chunk, .all),
+			.degradable => _pasteInGeneration(self, pos, chunk, .degradable),
+		}
+	}
+
+	fn _pasteInGeneration(self: Blueprint, pos: Vec3i, chunk: *ServerChunk, comptime mode: PasteMode) void {
+		const startX = pos[0];
+		const startY = pos[1];
+		const startZ = pos[2];
+
+		for(0..self.blocks.width) |x| {
+			const worldX = startX + @as(i32, @intCast(x));
+
+			for(0..self.blocks.depth) |y| {
+				const worldY = startY + @as(i32, @intCast(y));
+
+				for(0..self.blocks.height) |z| {
+					const worldZ = startZ + @as(i32, @intCast(z));
+					if(!chunk.liesInChunk(worldX, worldY, worldZ)) continue;
+
+					const block = self.blocks.get(x, y, z);
+					if(sbb.isOriginBlock(block) or sbb.isChildBlock(block) or block.typ == voidType) continue;
```

Comment:
Could we replace all origin and child blocks by void blocks during loading?
Then we would only need one comparison for each placed block, instead of 2 + a hashmap lookup.

Reply:
Ideally I would like this alongside `/set` being a method instead of function, with all the replacements being implemented as load time.

Reply:
I would also prefer to have EditMasks for that

Reply:
Alright, i suppose back to being draft until #1284 #1352 #1337 are done

Reply:
Generic way of replacing some blocks with different blocks

Reply:
But should we use the generic way? We already know where all the child and origin blocks are, why not just update them directly instead of iterating through all blocks in the volume again, running some generic (=expensive) test function on all blocks?

Reply:
Alright, maybe we should do it non generic way

Reply:
Okay, so only drawback is that rotated copies need to be created before BlueprintEntry.init because it removes origin and child blocks in place.

Reply:
Can't you replace them after you found all the child blocks?

Reply:
I could, but why bother, if I need to make those copies anyway. Code would get much more complicated if I had to actually rotate those blocks instead of just rediscovering them for every rotation. I also do It only once, while opening world, so again, why bother.